src/topology/algebra/ordered.lean

```diff
@@ -96,6 +96,28 @@ lemma closure_lt_subset_le [topological_space β] {f g : β → α} (hf : contin
   closure {b | f b < g b} ⊆ {b | f b ≤ g b} :=
 by { rw [←closure_le_eq hf hg], exact closure_mono (λ b, le_of_lt) }
 
+lemma continuous_within_at.closure_le [topological_space β]
+ {f g : β → α} {s : set β} {x : β} (hx : x ∈ closure s)
+ (hf : continuous_within_at f s x)
+ (hg : continuous_within_at g s x)
+ (h : ∀ y ∈ s, f y ≤ g y) : f x ≤ g x :=
+begin
+  let ψ := λ x, (f x, g x),
+  have cont : continuous_within_at ψ s x,
+    from continuous_within_at.prod hf hg,
+  let H :=  {p : α × α | p.1 ≤ p.2},
+  change ψ x ∈ H,
+  suffices : ψ x ∈ closure H,
+    by rwa ← (closure_eq_iff_is_closed.2 (ordered_topology.is_closed_le' α) : closure H = H),
+  rw [closure_eq_nhds, mem_set_of_eq] at *,
```

Comment:
I think you can take a shortcut by using `continuous_within_at.mem_closure_image` here. Possibly with the following version that could be worth adding just below `continuous_within_at.mem_closure_image`.
```lean
lemma continuous_within_at.mem_closure {f : α → β} {s : set α} {x : α} {A : set β}
  (h : continuous_within_at f s x) (hx : x ∈ closure s) (hA : s ⊆ f⁻¹' A) : f x ∈ closure A :=
closure_mono (image_subset_iff.2 hA) (h.mem_closure_image hx)
```

Reply:
Nice! I've done it.